lib/puppet/functions/gitlab_ci_runner/register.rb

```diff
@@ -0,0 +1,24 @@
+require_relative '../../../puppet_x/gitlab/runner.rb'
+
+# A function that registers a Gitlab runner on a Gitlab instance. Be careful, this will be triggered on noop runs as well!
```

Comment:
I'm going to ask around and see if we can't detect we're running in noop.

Reply:
Interesting idea 🤔 Would you suggest to just return dummy data in that case? Because the return data of that function is relevant for the Gitlab Runner config file.

Reply:
Ah 🙈 nvm.